File: src/prime.cpp
#include <iostream>
#include <math.h>
#include <time.h>
#include <cstdint>

bool is_prime(uint32_t n) {
    if (n <= 1) {
        return false;
    }
    
    uint32_t end = uint32_t(sqrt(n));
    for (uint32_t i = 2; i <= end; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    clock_t start = clock();
    uint32_t c = 0;
    for (uint32_t i = 0; i < 9000000; i++) {
        if (is_prime(i)) {
            c += 1;
        }
    }
    std::cout << c << std::endl;
    std::cout << (clock() - start)/1000 << "ms" << std::endl;
    return 0;
}

File: src/prime.py
import math
import time


def is_prime(n):
    if n % 2 == 0:
        return False

    end = int(math.sqrt(n))
    for i in range(3, end, 2):
        if n % i == 0:
            return False

    return True


start = int(time.time())
c = 1  # value is 1 becuase loop started from 3
for i in range(3, 9000000):
    if is_prime(i):
        c += 1

print("Process time: ", int(time.time()) - start, " second(s)")
print("Count of prime(s): ", c)
